src/circuit/commit_ivk.rs

```diff
@@ -822,34 +825,29 @@ mod tests {
                     rivk_gadget,
                 )?;
 
-                let expected_ivk = {
-                    let domain = CommitDomain::new(COMMIT_IVK_PERSONALIZATION);
-                    // Hash ak || nk
-                    domain
-                        .short_commit(
-                            iter::empty()
-                                .chain(
-                                    self.ak
-                                        .unwrap()
-                                        .to_le_bits()
-                                        .iter()
-                                        .by_vals()
-                                        .take(L_ORCHARD_BASE),
+                self.ak
+                    .zip(self.nk)
+                    .zip(ivk.inner().value())
+                    .assert_if_known(|((ak, nk), ivk)| {
+                        let expected_ivk = {
+                            let domain = CommitDomain::new(COMMIT_IVK_PERSONALIZATION);
+                            // Hash ak || nk
+                            domain
+                                .short_commit(
+                                    iter::empty()
+                                        .chain(
+                                            ak.to_le_bits().iter().by_vals().take(L_ORCHARD_BASE),
+                                        )
+                                        .chain(
+                                            nk.to_le_bits().iter().by_vals().take(L_ORCHARD_BASE),
+                                        ),
+                                    &rivk,
                                 )
-                                .chain(
-                                    self.nk
-                                        .unwrap()
-                                        .to_le_bits()
-                                        .iter()
-                                        .by_vals()
-                                        .take(L_ORCHARD_BASE),
-                                ),
-                            &rivk,
-                        )
-                        .unwrap()
-                };
+                                .unwrap()
+                        };
 
-                assert_eq!(&expected_ivk, ivk.inner().value().unwrap());
+                        &&expected_ivk == ivk
```

Comment:
I was curious how the parsing ambiguity between `&` and boolean `&&` is resolved. This explains it: https://doc.rust-lang.org/reference/expressions/operator-expr.html#borrow-operators